mllib-tests/v1p4/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -138,11 +143,12 @@ class GLMRegressionTest(sc: SparkContext) extends GLMTests(sc) {
       throw new IllegalArgumentException(
         s"GLMRegressionTest run with unknown loss ($loss).  Supported values: l2.")
     }
-    if (!Array("none", "l1", "l2").contains(regType)) {
+    if (!Array("none", "l1", "l2", "elastic-net").contains(regType)) {
       throw new IllegalArgumentException(
-        s"GLMRegressionTest run with unknown regType ($regType).  Supported values: none, l1, l2.")
+        s"GLMRegressionTest run with unknown regType ($regType).  Supported values: none, l1, l2, elastic-net")
     }
-    if (!Array("sgd").contains(optimizer)) { // only SGD supported in Spark 1.1
+    if (!(Array("sgd").contains(optimizer) || Array("elastic-net").contains(regType))) {
```

Comment:
This breaks the previous check i.e. now I can set optimizer="bla" and regType="elastic-net" and things will work.

What about refactoring 146-154 to check the combination of optimizer and regularization is either (sgd, {none, l1, l2}) or (lbfgs, elastic-net) and reporting error appropriately (e.g. sgd can only be used with regTypes ..., lbfgs can only be used with elastic-net regularization)